[Feature] SSD Shell 의 read 기능 추가 구현

Ellen99999999 opened the pull request:
변경점 : ssd_output.txt 에서 lba(address) 에 해당하는 값을 읽어서 string 을 뽑아 shell 응답 값을 만들어줌

검증

[ v] Build

[ v] Regression Test Pass

[ v] 해당 기능 TC

Ellen99999999 committed:
[Feature] SSD Shell 의 read 기능 추가 구현

변경점 : ssd_nand.txt 에서 lba(address) 에 해당하는 값을 읽어서 string 을 뽑아 shell 응답 값을 만들어줌

검증

[ v] Build

[ v] Regression Test Pass

[ v] 해당 기능 TC

Ellen99999999 committed:
[Feature] ssd shell 에서 ssd mock 을 통해서 읽는 동작 구현 수정

변경점 : ssd driver 추가함

검증

[v] Build

[v] Regression Test Pass

[v] 해당 기능 TC

flowinto10 commented on TeamBest_SSD_Shell/SSD_Shell.cpp:
```diff
@@ -1,10 +1,40 @@
 using namespace std;
 
-class SSDShell : public IShell{
+class SSDDriver {
```
드라이버를 분리하셨군요? SSD Read/Write 동작만 분리하기 위함일까요?

Ellen99999999 replied:
네네 맞습니다.

flowinto10 reviewed (approved):
고생하셨습니다.

youngchul-hwang commented on TeamBest_SSD_Shell/SSD_Shell.cpp:
```diff
@@ -17,9 +47,35 @@ class SSDShell : public IShell{
+		ssdDriver.read(address);
+
+		ifstream inputFile(SSD_NAND);
+		string buffer;
```
요구사항 명세를 보면 
ssd_nand.txt가 아니라 ssd_output.txt에서 데이터를 가져와야 하는 것 같아요.

Ellen99999999 replied:
감사합니다. 제가 중요한걸 잊고 있었네요. 업데이트 하겠습니다.

youngchul-hwang reviewed (commented):
수고하셨습니다.
요구사항에 맞지 않는 구현이 있는 것 같네요.
확인 부탁 드립니다.

Ellen99999999 committed:
[Feature or Refactor] 해당 Feature

변경점

검증

[ v] Build

[ v] Regression Test Pass

[ v] 해당 기능 TC

Ellen99999999 committed:
[Feature] ssd_output.txt 에서 lba(address) 에 해당하는 값을 읽어서 string 을 뽑아 shell 응답 값을 만들어줌

변경점 : 요구사항 잘못 이해하여 ssd_nand.txt 에서 읽었던것을 ssd_output.txt 를 통해 읽는 것으로 변경함

검증

[ v] Build

[ v] Regression Test Pass

[ v] 해당 기능 TC

Ellen99999999 committed:
[Feature] ssd_output.txt 에서 lba(address) 에 해당하는 값을 읽어서 string 을 뽑아 shell 응답 값을 만들어줌 (TC 업데이트)

변경점 : TC 에서 ssd_output.txt 에 expected, actual 을 구분하여 동작할 수 있도록 수정함
검증

[ v] Build

[ v] Regression Test Pass

[ v] 해당 기능 TC

R1NRINRIN reviewed (approved):
고생 많으셨습니다

youngchul-hwang reviewed (approved):
깔끔한 코드네요. 수고하셨습니다.
